### PR TITLE
[Serializer] convert empty CSV header names into numeric keys

### LINKED ISSUE #54709: Fatal error in CsvEncoder
### Symfony version(s) affected

7.0.6

### Description

Fuzzing reveals that CsvEncoder throws a fatal error with some values:
```
PHP Fatal error:  Uncaught TypeError: Cannot access offset of type string on string in /home/pierre/www/fuzzer/vendor/symfony/serializer/Encoder/CsvEncoder.php:179
```

### How to reproduce

Here is the smallest value I was able to reproduce with:
```php
<?php

require __DIR__.'/vendor/autoload.php';

$encoder = new \Symfony\Component\Serializer\Encoder\CsvEncoder();
var_dump($encoder->decode(',.
,', 'csv'));
```

### Possible Solution

_No response_

### Additional Context

I am using [php-fuzzer](https://github.com/nikic/PHP-Fuzzer) to test various input.

### PR DESCRIPTION
**Pull request by xabbuh**
| Q             | A
| ------------- | ---
| Branch?       | 5.4
| Bug fix?      | yes
| New feature?  | no
| Deprecations? | no
| Issues        | Fix #54709
| License       | MIT

### CODE REVIEW

File: src/Symfony/Component/Serializer/Encoder/CsvEncoder.php [hunk 1]
```diff
@@ -181,18 +181,24 @@ public function decode(string $data, string $format, array $context = [])
                 $depth = $headerCount[$i];
                 $arr = &$item;
                 for ($j = 0; $j < $depth; ++$j) {
+                    $headerName = $headers[$i][$j];
+
+                    if ('' === $headerName) {
```
Comment (xabbuh) [inline src/Symfony/Component/Serializer/Encoder/CsvEncoder.php:186]: I am not sure that we should really allow getting to this place at all. Above the outer `for` loop we check the first line of the CSV to extract headers names. Would it maybe be better to throw a meaningful exception there if empty header names are used?

Comment (dunglas) [reply src/Symfony/Component/Serializer/Encoder/CsvEncoder.php:186]: Both solutions look ok to me.

Comment (nicolas-grekas) [reply src/Symfony/Component/Serializer/Encoder/CsvEncoder.php:186]: Not sure about throwing. It's better to do our best to parse any doc IMHO.

**Review by nicolas-grekas — APPROVED** [review]

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @xabbuh.

**Pull request merged**